processor/src/main/java/org/mapstruct/ap/internal/conversion/PossibleNumberToStringConversion.java

```diff
@@ -0,0 +1,168 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.internal.conversion;
+
+import org.mapstruct.ap.internal.model.common.ConversionContext;
+import org.mapstruct.ap.internal.model.common.Type;
+import org.mapstruct.ap.internal.util.NativeTypes;
+
+
+import org.mapstruct.ap.internal.util.Strings;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.text.DecimalFormat;
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Conversion hooks for {@link Number} to {@link String}. The idea being that <em>if</em>
+ * we have a conversion to string and the source type is a {@code Number} and the mapping
+ * has a {@link org.mapstruct.Mapping#numberFormat()} we will replace the standard conversion
+ * with a {@link DecimalFormat} version
+ *
+ * @author Ciaran Liedeman
+ */
+public abstract class PossibleNumberToStringConversion extends SimpleConversion {
```

Comment:
I see. Maybe we should separate boolean from the numeric ones then? Not strictly needed, but a bit cleaner IMO.